js/src/ui/Form/AddressSelect/addressSelect.js

```diff
@@ -15,262 +15,799 @@
-    };
+    window.addEventListener('resize', this.handleTagsOpacity);
+  }
 
-    return { autocompleteEntries, entries };
+  componentDidMount () {
+    this.setPosition();
   }
 
-  shouldComponentUpdate (nextProps, nextState) {
-    const keys = [ 'error', 'value' ];
+  componentWillReceiveProps (nextProps) {
+    if (this.values && this.values.length > 0) {
+      return;
+    }
 
-    const prevValues = pick(this.props, keys);
-    const nextValues = pick(nextProps, keys);
+    this.setValues(nextProps);
+  }
 
-    return !isEqual(prevValues, nextValues);
+  componentWillUnmount () {
+    window.removeEventListener('resize', this.handleTagsOpacity);
   }
 
-  componentWillMount () {
-    const { value } = this.props;
-    const { entries, autocompleteEntries } = this.entriesFromProps();
-    const addresses = Object.keys(entries).sort();
+  setValues (props = this.props) {
+    const { accounts = {}, contracts = {}, contacts = {}, wallets = {} } = props;
 
-    this.setState({ autocompleteEntries, entries, addresses, value });
-  }
+    const accountsN = Object.keys(accounts).length;
+    const contractsN = Object.keys(contracts).length;
+    const contactsN = Object.keys(contacts).length;
+    const walletsN = Object.keys(wallets).length;
 
-  componentWillReceiveProps (newProps) {
-    if (newProps.value !== this.props.value) {
-      this.setState({ value: newProps.value });
+    if (accountsN + contractsN + contactsN + walletsN === 0) {
+      return;
     }
+
+    this.values = [
+      {
+        label: 'accounts',
+        values: [].concat(
+          Object.values(wallets),
+          Object.values(accounts)
+        )
+      },
+      {
+        label: 'contacts',
+        values: Object.values(contacts)
+      },
+      {
+        label: 'contracts',
+        values: Object.values(contracts)
+      }
+    ];
+
+    this.handleChange();
   }
 
   render () {
-    const { allowInput, disabled, error, hint, label } = this.props;
-    const { autocompleteEntries, value } = this.state;
+    const input = this.renderInput();
+    const content = this.renderContent();
+
+    const classes = [ styles.main ];
+
+    return (
+      <div
+        className={ classes.join(' ') }
+        onBlur={ this.handleMainBlur }
+        onClick={ this.handleFocus }
+        onFocus={ this.handleMainFocus }
+        onKeyDown={ this.handleInputAddresKeydown }
+        ref='inputAddress'
+        tabIndex={ 0 }
+      >
+        { input }
+        { content }
+      </div>
+    );
+  }
 
-    const searchText = this.getSearchText();
-    const icon = this.renderIdentityIcon(value);
+  renderInput () {
+    const { focused } = this.state;
+    const { accountsInfo, disabled, error, hint, label, value } = this.props;
+
+    const input = (
+      <InputAddress
+        accountsInfo={ accountsInfo }
+        error={ error }
+        focused={ focused }
+        label={ label }
+        hint={ hint }
+        tabIndex={ -1 }
+        value={ value }
+
```

Comment:
Extra line, non-alphabetical order (especially with this amount - keep it maintainable)